release/ray_release/job_manager/anyscale_job_manager.py

```diff
@@ -259,6 +270,54 @@ def run_and_wait(
         )
         return self._wait_job(timeout)
 
+    def get_last_ray_error_logs(self) -> Optional[str]:
+        globs = [
+            "dashboard.log",
+            "dashboard.err",
+            "dashboard_agent.log",
+            "runtime_env_agent.log",
+            "raylet.out",
+            "raylet.err",
+            "gcs_server.out",
+            "gcs_server.err",
+        ]
+        for glob in globs:
+            last_ray_logs = self._get_last_ray_error_logs(
+                self.cluster_manager.cluster_id,
+                glob,
+            )
+            if last_ray_logs:
+                return last_ray_logs
+        return None
+
+    def _get_last_ray_error_logs(self, cluster_id: int, glob: str) -> Optional[str]:
+        logs_controller = LogsController()
+        filter = LogFilter(
+            cluster_id=cluster_id,
+            glob=glob,
+            node_type=NodeType.HEAD_NODE,
+        )
+        log_group = logs_controller.get_log_group(
+            filter=filter,
+            page_size=DEFAULT_PAGE_SIZE,
+            timeout=timedelta(seconds=DEFAULT_TIMEOUT),
+            ttl_seconds=DEFAULT_TTL,
+        )
+        buf = io.StringIO()
```

Comment:
can these logs be big? maybe we should create a temp directory, download the logs into the directory (which is better supported in the SDK), and then postprocess them (find ERROR?) in some way?